### PR TITLE
#231 Migrate TabbedPane and Toolbar to JavaFX

### PR DESCRIPTION
**Pull request by kkutschera**
Summary of Changes:
- Stylesheet UMLEditorStyle.css added to UMLEditor
- Removed WelcomeButtonUI since its functionality is replaced by the stylesheet
- Removed ButtonTabComponent and VerticalLayout since not need for JavaFX nodes
- Extraneous `Plaform.runLater` calls have been removed from EditorFrame
- `aTabbedPane` converted to a JavaFX TabPane
- GraphPanel extends Tab that contains the JavaFX toolbar and a SwingNode containing the GraphFrame
- modified setTitle() in GraphPanel to always handle setting the title of its tab (no longer split between itself and the EditorFrame)
- GraphPanel holds a reference to its GraphFrame since it is no longer accessible through the object graph
- Multiple tests updated to fulfill the requirements of the new GraphPanel constructor
- Added `new JFXPanel()` in TestUsageScenariosStateDiagram to initialize JavaFX toolkit in the test
- WelcomeTab extends Tab and simplified the layout to a left panel, right panel, and a footer

- Toolbar extends BorderPane whose contents will be a vertical FlowPane of diagram tools and a button to either expand or collapse the toolbar
- Toolbar retains some Swing components to allow the use of the popup/context menu on the GraphPanel (this will easily be removed once the GraphPanel has been migrated to JavaFX)
- ImageCreator replaces the functionality of IconCreator in JavaFX (used by Toolbar)
- `freeCtrlTab` in Toolbar has been removed since CTRL+TAB is the default for cycling through windows in JavaFX
- EditorFrame instance removed since now accessible to Toolbar through the object graph

### CODE REVIEW

File: src/ca/mcgill/cs/jetuml/gui/EditorFrame.java [hunk 1]
```diff
@@ -445,18 +412,15 @@ public void addGraphType(String pResourceName, final Class<?> pGraphClass)
...
-				}
-				catch (Exception exception) 
-				{
-					exception.printStackTrace();
-				}
-			});
+				GraphFrame frame = new GraphFrame((Graph) pGraphClass.newInstance(), aTabbedPane);
+				addTab(frame);
+			}
+			catch (Exception exception)
```
Comment (prmr) [inline src/ca/mcgill/cs/jetuml/gui/EditorFrame.java]: should be narrowed to precise type: `IllegalAccessException`. Please remove excess white space and lines. Replace `printStackTrace` with `assert false;`

File: src/ca/mcgill/cs/jetuml/gui/EditorFrame.java [hunk 2]
```diff
@@ -684,13 +572,10 @@ private void buildRecentFilesMenu()
    		{
    			String name = "_" + i + " " + file.getName();
    			final String fileName = file.getAbsolutePath();
-   			aRecentFilesMap.put(name.substring(3), pEvent -> open(fileName));
+   			aRecentFilesMap.put(name.substring(3), pEvent -> open(fileName));
```
Comment (prmr) [inline src/ca/mcgill/cs/jetuml/gui/EditorFrame.java]: why the extra spaces?

File: src/ca/mcgill/cs/jetuml/gui/EditorFrame.java [hunk 3]
```diff
@@ -819,27 +703,24 @@ public void close()
 		{
 			return;
 		}
-		JInternalFrame curFrame = (JInternalFrame) aTabbedPane.getSelectedComponent();
+		Tab curFrame = (Tab) aTabbedPane.getSelectionModel().getSelectedItem();
 		if (curFrame != null)
```
Comment (prmr) [inline src/ca/mcgill/cs/jetuml/gui/EditorFrame.java]: please rename `currentFrame`

File: src/ca/mcgill/cs/jetuml/gui/GraphFrame.java [hunk 4]
```diff
@@ -21,40 +21,59 @@
...
 	/**
      * Constructs a graph frame with an empty tool bar.
      * @param pGraph the initial graph
-     * @param pTabbedPane the JTabbedPane associated with this GraphFrame.
+     * @param pTabbedPane the TabPane associated with this GraphFrame.
 	 */
-	public GraphFrame(Graph pGraph, JTabbedPane pTabbedPane)
+	public GraphFrame(Graph pGraph, TabPane pTabbedPane)
 	{
+		super();
```
Comment (prmr) [inline src/ca/mcgill/cs/jetuml/gui/GraphFrame.java]: extraneous super consts call, can be removed

File: src/ca/mcgill/cs/jetuml/gui/GraphFrame.java [hunk 5]
```diff
@@ -21,40 +21,59 @@
...
 		aTabbedPane = pTabbedPane;
 		ToolBar sideBar = new ToolBar(pGraph);
-		aPanel = new GraphPanel(pGraph, sideBar);
-		Container contentPane = getContentPane();
-		contentPane.add(sideBar, BorderLayout.EAST);
-		contentPane.add(new JScrollPane(aPanel), BorderLayout.CENTER);
-		setComponentPopupMenu( null ); // Removes the system pop-up menu full of disabled buttons.
+		aPanel = new GraphPanel(pGraph, sideBar, this);
+
```
Comment (prmr) [inline src/ca/mcgill/cs/jetuml/gui/GraphFrame.java]: extra line

File: src/ca/mcgill/cs/jetuml/gui/GraphPanel.java [hunk 6]
```diff
@@ -94,6 +93,7 @@
 	private static final Color GRABBER_FILL_COLOR = new Color(173, 193, 214);
 	private static final Color GRABBER_FILL_COLOR_TRANSPARENT = new Color(173, 193, 214, 75);
 	
+	private GraphFrame aFrame;
```
Comment (prmr) [inline src/ca/mcgill/cs/jetuml/gui/GraphPanel.java]: doesn't this introduce a circular dependency?

Comment (kkutschera) [reply src/ca/mcgill/cs/jetuml/gui/GraphPanel.java]: Once `GraphPanel` is migrated to JavaFX, the `GraphFrame` will be accessible through the object graph and the circular dependency will be removed.

**Pull request merged**